src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -273,6 +296,31 @@ def global_datasymbols(self):
         '''
         return [sym for sym in self.datasymbols if sym.is_global]
 
+    @property
+    def unresolved_datasymbols(self):
+        '''
+        :returns: list of symbols that have an unknown interface (i.e. we \
+                  don't know how they are brought into scope).
```

Comment:
Should "brought into scope" be "declared"?

Reply:
I don't think so. A "use my_mod, only: my_var" doesn't declare my_var.

Reply:
True, but an unresolved data symbol does not have to be brought from a use statement, it could be declared later, or in the parent container. However, on reflection I think it could be argued that declaring a variable is essentially bringing it into scope so I'm happy.